doc/source/mergingdata.rst

```diff
@@ -1,15 +1,73 @@
+.. currentmodule:: geopandas
+
+.. ipython:: python
+   :suppress:
+
+   import geopandas as gpd
+   world = gpd.GeoDataFrame().from_file('_example_data/naturalearth_lowres.shp')
+   cities = gpd.GeoDataFrame().from_file('_example_data/naturalearth_cities.shp')
+   
+   # For attribute join
+   country_shapes = world[['geometry', 'iso_a3']]
+   country_names = world[['name', 'iso_a3']]
+
+   # For spatial join
+   countries = world[['geometry', 'name']]
+   countries = countries.rename(columns={'name':'country'})
+
+
 
 Merging Data
 =========================================
 
+There are two ways to combine datasets in *geopandas* -- attribute joins and spatial joins.
+
+In an attribute join, a ``GeoSeries`` or ``GeoDataFrame`` is combined with a regular *pandas* ``Series`` or ``DataFrame`` based on a common variable. This is analogous to normal merging or joining in *pandas*.
+
+In a Spatial Join, observations from to ``GeoSeries`` or ``GeoDataFrames`` are combined based on their spatial relationship to one another. 
 
 Attribute Joins
 ----------------
 
-[TO BE COMPLETED -- EXAMPLES OF JOINING GDF WITH PANDAS DATAFRAME]
+Attribute joins are accomplished using the ``merge`` method. Note that unlike in *pandas*, ``merge`` is a method called from the spatial dataset, not a stand-alone function. 
```

Comment:
> Note that unlike in *pandas*, ``merge`` is a method called from the spatial dataset, not a stand-alone function

This is not fully correct, as in pandas you have both the method and the top-level function. So I would just emphasize that you need to use the method (to have a GeoDataFrame as the result)